### ISSUE TITLE
Can removeAttrs preserve currentColor like grunt-svgstore does?

### ISSUE THREAD

**Issue by roblevintennis** [svg/svgo#999]
@GreLI I'm not sure if this is related to https://github.com/svg/svgo/issues/967 as that's for `convertColor` and mine is really for `removeAttrs`...

I'm transitioning our SVG build process from stand alone `grunt-svgstore` to now use Webpack `svg-sprite-loader` and hoping to do the cleansing with svgo. There was a feature we leaned on heavily (which I contributed in fact):

>Using the value of currentColor on any property with the key `fill`, will result in that property remaining in the resulting SVG (regardless of whether or not you ask for fill to be cleaned via cleanup). This can be used to achieve accent color for you SVG instances by defining the font color on a parent element via the CSS color property.

https://github.com/FWeinb/grunt-svgstore#supplemental-features
https://github.com/FWeinb/grunt-svgstore/pull/63

In their case, you could then use the following config option: `cleanup: ['fill', 'stroke'],` .. I'd like to do something similar like:
`{removeAttrs: {attrs: "(fill|stroke)"}}`. However, it appears this would result in any and all fills/strokes being removed regardless. Is there a way and/or would this be something svgo would like contributed if not? Also, if there's another webpack plugin you know of that does this already that could work for me too. Obviously, I'm willing to contribute if I'm capable :)

UPDATE:

Yeah, I would think adding this optional param to removeAttrs and have it default to opting out would preserve your current behavior but at least allow such override:

```javascript
exports.params = {
  elemSeparator: DEFAULT_SEPARATOR,
  preserveCurrentColor: false,
  attrs: []
}
```

I'll try to code up a proof of concept if I can get a spare cycle

**Comment by Herman-Freund** [comment]
Would #977 help?

P.S. The removeAttrs plugin is meant to deal with any attribute (and not just the ones dealing with colors).

**Comment by roblevintennis** [comment]
@Herman-Freund thanks for referencing that PR. Gee, it doesn't look like it's got a code review yet 😕 

I could certainly achieve it if `value` is exposed in the regex filtering `{"attrs":"*:(stroke|fill):((?!^currentColor$).)*"}` if I'm understanding right?

Using currentColor to inherit the font color is such a popular SVG trick, I wonder if my PR should still be used additionally. Crafting a negative regex might be a bit to ask of a more neophyte user of this plugin (although I suppose the examples help).

Anyway, I'll plus your PR

**Comment by Herman-Freund** [comment]
> I could certainly achieve it if value is exposed in the regex filtering {"attrs":"*:(stroke|fill):((?!^currentColor$).)*"} if I'm understanding right?

Correct

> I wonder if my PR should still be used additionally

I think it should be somewhat merged (just brainstorming) e.g.
```typescript
// simple mode
removeAttrs: string|RegEx

// full feature mode
removeAttrs: [
  {
    attribute: string|RegEx
    element?: string|RegEx
    value?: string|RegEx
  }
]
```
But this is basically redoing the entire plugin (and will probably break users existing code).

> Anyway, I'll "emoji plus" your PR

👍 Thanks!

**Comment by SethFalco** [comment]
Looks like this was addressed in #1000 already, but this PR just wasn't closed when it was merged.

Thanks for filing the report and contributing the functionality you needed. Welcome to let me know if I've misunderstood the status of this request. 👍🏽
